dbms/src/Common/HashTable/StringHashTable.h

```diff
@@ -296,13 +295,19 @@ class StringHashTable : private boost::noncopyable
             if ((reinterpret_cast<uintptr_t>(p) & 2048) == 0)
             {
                 memcpy(&n[0], p, 8);
-                n[0] &= -1ul >> s;
+                if constexpr (DB::isLittleEndian())
+                    n[0] &= (-1ULL >> s);
+                else
+                    n[0] &= (-1ULL << s);
             }
             else
             {
                 const char * lp = x.data + x.size - 8;
                 memcpy(&n[0], lp, 8);
-                n[0] >>= s;
+                if constexpr (DB::isLittleEndian())
```

Comment:
Whether the same fix is needed in TwoLevelStringHashTable?

Reply:
Good catch, I'll apply it also.